fabric-level-generators-v1/src/main/java/net/fabricmc/fabric/mixin/level/generator/MixinGameJoinS2CPacket.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.level.generator;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.client.network.packet.GameJoinS2CPacket;
+import net.minecraft.world.level.LevelGeneratorType;
+
+@Mixin(GameJoinS2CPacket.class)
+public class MixinGameJoinS2CPacket {
+	@Redirect(method = "write", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/level/LevelGeneratorType;getName()Ljava/lang/String;"))
+	private String changeSentLevelGeneratorType(LevelGeneratorType levelGeneratorType) {
+		return LevelGeneratorType.DEFAULT.getName();
```

Comment:
Like coderbot said as I was typing this, you should probably only change this to DEFAULT when the generator type is not vanilla. Even better, detect if the client has the generator type registered, although I'm not sure whether that's possible, so ya just assume the client doesn't know about the modded ones.

Reply:
We don't want to send modded level types over the network because there is a strict length limit of 16 chars which is barely enough for many mod IDs. The level type is apparently only used for crash reports and the snooper anyways so it isn't a big deal to sync.

Reply:
I mean you could just increase that limit... But probably not worth it